### PR TITLE
[ADD] email marketing: analyze metrics

### PR DESCRIPTION
**Pull request by jero-odoo**
Creating a new doc to outline how to analyze metrics in the email marketing app
Docs task: https://www.odoo.com/mail/view?model=project.task&res_id=3894022&access_token=d13dbf1e-c5cd-4052-92ec-c8031481e5b9

### CODE REVIEW

**Comment by jero-odoo** [comment]
Hey @larm-odoo this one is ready for your review. Thanks!

**Review by larm-odoo — COMMENTED** [review]
Nice new doc! I love it- it's great info to share. I had just a general suggestion, and then some questions on the app itself/what can be included. Might be worth a suggestion to the product owner as well (I didn't see any 'unsubscribe' numbers, or 'report as spam'- which I think is important). Ping me again when you're done, just because I am curious to the answers, and would love to read it again.

File: content/applications/marketing/email_marketing/analyze_metrics.rst [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+      *Bounced rate* measures the number of messages that were unsuccessfully delivered, against the
+      number that were sent in total. Similarly to *received rate*, this metric can be used to
+      judge the integrity of the addresses on an email list, and can be used to indicate quality
+      issues. A higher bounced rate suggests the list may be out-of-date, or the addresses are
+      incorrect.
+
+.. image:: analyze_metrics/mailings-metrics.png
+   :align: center
+   :alt: The results of sent mailings in the email marketing application.
+
```
[the remark this answers is not in the document]
Comment (jero-odoo) [reply content/applications/marketing/email_marketing/analyze_metrics.rst:62]: There is another doc that handles that (https://www.odoo.com/documentation/17.0/applications/marketing/email_marketing/unsubscriptions.html)

File: content/applications/marketing/email_marketing/analyze_metrics.rst [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+Bounced rate
+~~~~~~~~~~~~
+
+Similarly to received rate, bounced rate can be used to measure the quality of the mailing list used
+in a campaign. In this case, a high bounce rate indicates potential issues with the quality or
+accuracy of the email list. It may indicate that there are many invalid or outdated email addresses
+on the list.
+
+Combining metrics
```
[the remark this answers is not in the document]
Comment (jero-odoo) [reply content/applications/marketing/email_marketing/analyze_metrics.rst:135]: again, there is another doc on the unsubscribe/blacklisting. I will add a seealso link to it :+1:

**Comment by jero-odoo** [comment]
This is ready for tech/final. Thanks!

**Review by samueljlieber — CHANGES_REQUESTED** [review]
Hi @jero-odoo, thank you for this addition to the email marketing documentation! I am excited to see more content being added here, especially in regards to reporting/analyzing mass mailings. 

While this is a great start on analyzing email marketing metrics, there is some duplicate content in this doc, some strong assumptions made about mailings, and some missing information.

I also ended up with several suggestions around restructuring the content in this doc and found it easier to just make a commit with these changes. So please find the second commit with my changes for your consideration.

Here is a brief summery of my changes and suggestions:
- Remove the **Key metrics** section and combine the metric definitions with each in the **Analyze metrics** section. 
- Remove **Individual metrics** heading and move each metric heading up a heading level.
- Make less definite assumptions as to why an email may have a specific open, replied, clicked, received, or bounced rate as there are many reasons for each. Instead, I suggest using terms like "may indicate" or "could suggest".
- Add a **Deliverability issues** section to detail technical reasons for a high bounce rate or low received rate.
- Add sections for the following metrics that are missing: Sent, Ignored, and Link Tracker. (Bonus probably in another PR in the future: Opportunities, Quotations, and Invoiced?)
- I think the **Combining metrics** is a cool idea, but due to the many possible reasons for each combination– I think it is best to not include this section, sticking to the individual metric definitions instead.

I understand these are a lot of changes, but I don't feel that this doc in it's current state is ready to go live for my reasons above. Please let me know your thoughts and if you want to workshop anything, I'm more than happy to!

Please tag me for another look once you revise this PR, thanks for your continued hard work on this 🙏

**Comment by jero-odoo** [comment]
Hey @samueljlieber thank you for all the work you did on this, I just had to make a few tweaks.
I understand wanting to limit the content to the barest essentials. However, I was hoping to include a few samples of the combined metrics to show how the varied results can be interpreted. And while yes, it is obviously based on assumptions more than hard facts, I don't think that would be out of line. I am fine merging it the way it is now, but I would like to consider expanding it in the future if there is a way to demonstrate that and focus on more potential real-life use cases. I would be happy to workshop that further. For now, I am ok with going forward as is.

Also:
-Add sections for the following metrics that are missing: Sent, Ignored, and Link Tracker. (Bonus probably in another PR in the future: Opportunities, Quotations, and Invoiced?)
Sent I don't know that I would consider this a metric perse, since it is simply a data point.
Ignored would be interesting to include, but I am not sure the best place to utilize it in the app
Link tracker specifically, I think, would be interesting on its own since there are a lot of avenues to explore with it.

**Review by samueljlieber — APPROVED** [review]
Thank you @jero-odoo for taking my suggestions into consideration, this PR is looking great! I am also comfortable merging as-is, but I agree with you there are a lot of opportunities here to improve and expand, especially with use-cases.

> I was hoping to include a few samples of the combined metrics to show how the varied results can be interpreted. And while yes, it is obviously based on assumptions more than hard facts, I don't think that would be out of line.

I completely agree, I think some assumptions have to be made. I just think we need to workshop framing the use-cases a little more in order to provide relevant examples. This should be a fun [IMP] to make in the near future 🙂

> Add sections for the following metrics that are missing: Sent, Ignored, and Link Tracker.

I believe **Sent** and **Ignored** are important to include since it defines the _total_ at which the other metrics are measured by, as well the **Ignored** smart button takes you to the Mailing Traces which show **Status** and **Failure Type** reasons for why a contact wasnt included in the mailing. See [this code](https://github.com/odoo/odoo/blob/855dc878a2c8abc594fc01d244bac10139b61702/addons/mass_mailing/models/mailing.py#L338) for how _total_ is calculated. 

I suppose **Link Tracker** could be its own doc possibly?

Either way, lets merge as is and note this for improvements down the road 🙂 👍
.....
@robodoo r+

**Pull request closed without merging**